### PR TITLE
Content Creator: Make module description and title mandatory

### PR DESCRIPTION
**Pull request by MajoDurco**
Solves #301.

### CODE REVIEW

File: preupg/creator/ui_helper.py [hunk 1]
```diff
@@ -356,12 +355,19 @@ def is_valid_string(string):
...
+    def ask_for_string(msg, err_msg):
+        string = get_user_input(msg, any_input=True)
+        if not UIHelper.is_valid_string(string):
             print(err_msg)
             return False
-        return version
+        return string
+
+    @staticmethod
+    def ask_mandatory_string(msg, err_msg):
```
Comment (bocekm) [inline preupg/creator/ui_helper.py]: Please use this cool new function also within specify_upgrade_path(). Thanks.

**Comment by AloisMahdal** [comment]
qa_ack: manual test should be enough

...although this will actually also heavily affect most of downstream preupgrade-assistant tests (tier "pa-r"), as these use fake modules created by this code.  So once the tests break, are updated and then start working again, we know this has been fixed.

**Comment by pirat89** [comment]
I recommend even replace the *content* by the *module* keyword as the *content* is relic of previous terminology that has been changed time ago. It can be done in separate PR as it will require even change inside modules. But I prefer to change it for planning release yet as we have more changes in modules already accepted.

**Comment by bocekm** [comment]
@pirat89, you mean changing the _content_ to _module_ in the tool name _preupg-content-creator_? Because there's still so many places where _content_ is used and it would be great to do one PR to replace this word everywhere.
BTW, this PR is now waiting for the QA test and then it will be merged.

**Comment by AloisMahdal** [comment]
Now, IIUC, p-c-c asks Y/N whether we want to provide *description*, but title is already mandatory.  So how should this PR affect the title?

**Comment by bocekm** [comment]
@AloisMahdal, the title was mandatory even before, truth to that. But not entering the title by user was handled by a separate function. Here, @MajoDurco used his new function `ask_mandatory_string` that is common for all mandatory inputs, and instead of exiting the tool when the title is not entered, it tells the user that the title is mandatory and asks the user for it again.

**Comment by AloisMahdal** [comment]
Actually it used to accept empty string for both title and description.  I have modified downstream test library to "make up" both strings if test does not provide them and also removed the `y` step.

**Comment by AloisMahdal** [comment]
Verified with downstream test *riskceck* and  preupgrade-assistant-2.4.4-0.201709121548Z.PR305.el6_8.

After mentioned changes to test library, test works fine.

**Pull request merged**